Lib/glyphsLib/builder/axes.py

```diff
@@ -157,8 +159,24 @@ def to_designspace_axes(self):
         ):
             axis_wanted = True
 
+        # See https://github.com/googlefonts/glyphsLib/issues/568
+        if custom_mapping:
+            if axis.tag in custom_mapping:
+                mapping = {float(k): v for k, v in custom_mapping[axis.tag].items()}
+                regularDesignLoc = axis_def.get_design_loc(regular_master)
+                reverse_mapping = {v: k for k, v in mapping.items()}
+                if regularDesignLoc in reverse_mapping:
+                    regularUserLoc = reverse_mapping[regularDesignLoc]
+                else:
+                    regularUserLoc = min(mapping)
+            else:
+                logger.warning(
+                    f"Skipping {axis.tag} since it hasn't been defined "
+                    "in the Axis Mapping"
+                )
+                continue
         # See https://github.com/googlefonts/glyphsLib/issues/280
-        if font_uses_new_axes(self.font):
+        elif font_uses_new_axes(self.font):
```

Comment:
i suggest renaming `font_uses_new_axes` to `font_uses_axis_locations` as that method is no longer "new", but de facto superseded by Axis Mappings approach